Microsoft.Azure.Cosmos/src/Resource/CosmosExceptions/CosmosExceptionFactory.cs

```diff
@@ -154,24 +159,36 @@ internal static class CosmosExceptionFactory
                 if (content != null
                && content.CanRead)
                 {
-                    try
+                    using (StreamReader streamReader = new StreamReader(content))
                     {
-                        Error error = Documents.Resource.LoadFrom<Error>(content);
-                        if (error != null)
+                        string errorContent = streamReader.ReadToEnd();
+                        try
                         {
-                            // Error format is not consistent across modes
-                            return (error, error.ToString());
+                            JObject errorObj = JObject.Parse(errorContent);
+                            Error error = errorObj.ToObject<Error>();
+                            if (error != null)
+                            {
+                                StringBuilder message = new StringBuilder();
+                                Dictionary<string, object> errorDictionary = errorObj.ToObject<Dictionary<string, object>>();
+                                foreach (KeyValuePair<string, object> entry in errorDictionary)
+                                {
+                                    message
+                                        .Append(Environment.NewLine)
+                                        .Append(entry.Key + " : " + entry.Value);
```

Comment:
Avoid the extra string allocation
```suggestion
                                        .Append(entry.Key)
                                        .Append(" : ")
                                        .Append(entry.Value);
```